src/com/qamp/HarisJasarevic/homeworks/homework5/task1/BankAccounts.java

```diff
@@ -0,0 +1,21 @@
+package com.qamp.HarisJasarevic.homeworks.homework5.task1;
+
+public abstract class BankAccounts {
+
+    protected AccountOwner accountOwner;
+    protected String serialNumber;
+    protected int passCode;
+    protected double balance;
+
+    public BankAccounts(final AccountOwner accountOwner, final String serialNumber, final int passCode) {
+        this.accountOwner = accountOwner;
+        this.serialNumber = serialNumber;
+        this.passCode = passCode;
+        this.balance = 0;
+    }
+
+    public abstract double moneyWithdraw ( final double withdrawAmount) throws IllegalStateException;
+    public abstract double moneyDeposit (final  double depositAmount) throws IllegalStateException;
```

Comment:
There is no need to mark these abstract methods with throws. It doesn't really matter in this case especially since it is a runtime exception.

But in case of a checked exception, every method that would override `moneyWithdraw` or `moneyDeposit`, must not throw a boarder exception, for example `RuntimeException`. Now whether that's a problem depends really on your use case.